### PR TITLE
Add boost/1.61.0/gcc-6.2-cmake-3.17.1 pkg, modulefile docs

### PR DESCRIPTION
**Pull request by willfurnass**

### CODE REVIEW

**Review by nikishe — APPROVED** [review]

File: sharc/software/modulefiles/libs/boost/1.61.0/gcc-6.2-cmake-3.17.1 [hunk 1]
```diff
@@ -0,0 +1,40 @@
+[sa_cs1wf@sharc-login1 boost]$ cat 1.61.0/gcc-6.2-cmake-3.17.1
```
Comment (jkwmoore) [inline sharc/software/modulefiles/libs/boost/1.61.0/gcc-6.2-cmake-3.17.1:1]: We might want to trim this line off :)

Comment (jkwmoore) [reply sharc/software/modulefiles/libs/boost/1.61.0/gcc-6.2-cmake-3.17.1:1]:
```suggestion
```

File: sharc/software/libs/boost.rst [hunk 2]
```diff
@@ -17,6 +17,7 @@ To activate the library run one of the following: ::
 
         module load libs/boost/1.64.0/gcc-4.9.4
         module load libs/boost/1.64.0/gcc-8.2-cmake-3.17
```
Comment (jkwmoore) [inline sharc/software/libs/boost.rst:19]: I bet I caused this error.

Comment (jkwmoore) [reply sharc/software/libs/boost.rst:19]:
```suggestion
        module load libs/boost/1.64.0/gcc-8.2-cmake-3.17.1
```

**Review by jkwmoore — APPROVED** [review]
LGTM.

**Pull request merged**